csw-server/src/main/java/org/fao/geonet/kernel/csw/services/getrecords/SearchController.java

```diff
@@ -97,21 +95,61 @@ public class SearchController {
+     * @param context                        service context
+     * @param id                             id of metadata
+     * @param setName                        requested ElementSetName
+     * @param outSchema                      requested OutputSchema
+     * @param elemNames                      requested ElementNames
+     * @param typeName                       requested typeName
+     * @param resultType                     requested ResultType
+     * @param strategy                       ElementNames strategy
+     * @param checkMetadataAvailableInPortal Checks if the metadata can be retrieved in the portal.
+     *                                       Used in GetRecordById. GetRecords does a query with this check already.
      * @return The XML metadata record if the record could be converted to the required output
      * schema. Null if no conversion available for the schema (eg. fgdc record can not be converted
      * to ISO).
+     * @throws CatalogException hmm
      */
-    public static Element retrieveMetadata(ServiceContext context, String id, ElementSetName setName, String
-        outSchema, Set<String> elemNames, String typeName, ResultType resultType, String strategy, String displayLanguage) throws CatalogException {
+    public Element retrieveMetadata(ServiceContext context, String id, ElementSetName setName, String
+        outSchema, Set<String> elemNames, String typeName, ResultType resultType, String strategy, String displayLanguage,
+                                    boolean checkMetadataAvailableInPortal) throws CatalogException {
+
+        if (checkMetadataAvailableInPortal) {
+            // Check if the metadata is available in the portal
+            String elasticSearchQuery = "{ \"bool\": {\n" +
+                "            \"must\": [\n" +
+                "        {" +
+                "          \"term\": {" +
+                "            \"id\": {" +
+                "              \"value\": \"%s\"" +
+                "            }" +
+                "          }" +
+                "        } " +
+                "            ]\n" +
+                "          ,\"filter\":{\"query_string\":{\"query\":\"%s\"}}}}";
+
+            JsonNode esJsonQuery;
+
+            try {
+                String filterQueryString = esFilterBuilder.build(context, "metadata", false, node);
+                String jsonQuery = String.format(elasticSearchQuery, id, filterQueryString);
+
+                ObjectMapper objectMapper = new ObjectMapper();
+                esJsonQuery = objectMapper.readTree(jsonQuery);
+
+                Set<String> fieldsToRetrieve = new HashSet<>();
+                fieldsToRetrieve.add("uuid");
+                SearchResponse result = searchManager.query(esJsonQuery, fieldsToRetrieve, 0, 1);
+
+                long numMatches = result.hits().hits().size();
+                if (numMatches == 0) {
+                    return null;
+                }
```

Comment:
We don't really need to retrieve the document, only a check that it is found in the portal should be good enough. So something like:

```suggestion
                TotalHits total = searchManager.query(esJsonQuery, new HashSet<>(), 0, 0).hits().total();

                if ((total != null ? total.value() : 0) == 0) {
                    return null;
                }
```

Also it is better to check the total than response hits size (same as in https://github.com/geonetwork/core-geonetwork/pull/7937) even if here it would work for 0 or 1 record.

Reply:
https://github.com/GeoCat/core-geonetwork/pull/100 with the import.